src/basic/string-util.h

```diff
@@ -33,6 +33,12 @@ static inline bool streq_ptr(const char *a, const char *b) {
         return strcmp_ptr(a, b) == 0;
 }
 
+static inline char* strstr_ptr(char *haystack, const char *needle) {
+        if (!haystack)
+                return NULL;
```

Comment:
hmm, i'd expect a function named like this to also be safe if needle is NULL. Should probably return NULL in that case?

Reply:
Hmm, or should it `assert(needle)`? I could see it both ways.

Reply:
Also, what should `streq_ptr(NULL, NULL)` return?

Reply:
I'd treat NULL as a special string that is not included in any other string, and thus we should return NULL here. And the special NULL/NULL case would also also return NULL, because NULL is so special it is not even included in the NULL string...

Reply:
Updated like that.